frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { useFormikContext, FormikValues } from 'formik';
+import { k8sGet } from '@console/internal/module/k8s';
+import { RoleBindingModel } from '@console/internal/models';
+import { checkAccess } from '@console/internal/components/utils';
+import { ResourceDropdownField } from '@console/shared';
+import { getProjectResource } from '../../../utils/imagestream-utils';
+import { ImageStreamActions as Action } from '../import-types';
+import { ImageStreamContext } from './ImageStreamContext';
+
+const ImageStreamNsDropdown: React.FC = () => {
+  const { setFieldValue, initialValues } = useFormikContext<FormikValues>();
+  const { dispatch } = React.useContext(ImageStreamContext);
+  return (
+    <ResourceDropdownField
+      name="imageStream.namespace"
+      label="Projects"
+      title="Select Project"
+      fullWidth
+      required
+      resources={getProjectResource()}
+      dataSelector={['metadata', 'name']}
+      onChange={(selectedProject) => {
+        const promiseArr = [];
+        setFieldValue('imageStream.image', '');
+        setFieldValue('imageStream.tag', '');
+        setFieldValue('isi', initialValues.isi);
+        dispatch({ type: Action.setLoading, value: true });
+        dispatch({ type: Action.setAccessLoading, value: true });
+        promiseArr.push(
+          checkAccess({
+            group: RoleBindingModel.apiGroup,
+            resource: RoleBindingModel.plural,
+            verb: 'create',
+            name: 'system:image-puller',
+            namespace: selectedProject,
+          })
+            .then((resp) =>
+              dispatch({ type: Action.setHasCreateAccess, value: resp.status.allowed }),
+            )
+            .catch(() => dispatch({ type: Action.setHasAccessToPullImage, value: false })),
+        );
+        promiseArr.push(
+          k8sGet(RoleBindingModel, 'system:image-puller', selectedProject)
```

Comment:
Getting 404 for this ```system:image-puller```. 

![Screenshot from 2020-01-03 14-07-32](https://user-images.githubusercontent.com/2561818/71714092-6ec70280-2e32-11ea-8dc4-1ba219d4bcbd.png)

Reply:
This api call to figure out whether we have access to grant permission, if the api throws 404 then catch block will set the grantAccess variable to false.

Reply:
Got it, Thanks.